hoomd/data/typeparam.py

```diff
@@ -34,6 +109,40 @@ def __setitem__(self, key, value):
         """Set parameters by key."""
         self.param_dict[key] = value
 
+    def __delitem__(self, key):
+        """__delitem__ is not available for `TypeParameter` objects."""
+        raise NotImplementedError("__delitem__ is not defined for this type.")
+
+    def get(self, key, value):
+        """Get values for keys with undefined keys returning default.
+
+        Args:
+            key:
+                Valid keys specifications (depends on the expected key length).
+            default (``any``, optional):
+                The value to default to if a key is not found in the mapping.
+                If not set, the value defaults to the mapping's default.
+
+        Returns:
+            values:
+                Returns a dict of the values for the keys asked for if multiple
```

Comment:
This can handle a list of keys? 

Or is this referring to ('A', 'B') "keys"?

Reply:
All allowed methods can handle the full indexing documented above in a mostly intuitive way (still no deleting of items allowed).